prow/gerrit/adapter/adapter.go

```diff
@@ -87,22 +87,24 @@ func NewController(lastSyncTracker LastSyncTracker, cookiefilePath string, proje
 // and creates prowjobs according to specs
 func (c *Controller) Sync() error {
 	syncTime := c.tracker.Current()
-	latest := syncTime
+	latest := syncTime.DeepCopy()
 
 	for instance, changes := range c.gc.QueryChanges(syncTime, c.config().Gerrit.RateLimit) {
 		for _, change := range changes {
 			if err := c.ProcessChange(instance, change); err != nil {
 				logrus.WithError(err).Errorf("Failed process change %v", change.CurrentRevision)
 			}
-			if latest.Before(change.Updated.Time) {
-				latest = change.Updated.Time
+			lastTime, ok := latest[instance][change.Project]
+			if !ok || lastTime.Before(change.Updated.Time) {
+				lastTime = change.Updated.Time
```

Comment:
Any chance there are paging issues?

AKA we look for changes since time X, it returns the first 10 out of 14, which we process. We then bump lastTime to 10.Updated.Time, which is after the last update for 11. So next loop our query misses 11-14.

Reply:
shouldn't hopefully. We paginate through individual Gerrit API calls, but we go through all changes in every sync loop anyway and the lastSync is only updated after the sync loop ends.